collector/sysctl_bsd.go

```diff
@@ -108,6 +109,29 @@ func (b bsdSysctl) Value() (float64, error) {
 		// This conversion maintains the usec precision.  Using
 		// the time package did not.
 		tmpf64 = unix + (usec / float64(1000*1000))
+	case bsdSysctlTypeCLong:
+		raw, err := unix.SysctlRaw(b.mib)
+		if err != nil {
+			return 0, err
+		}
+
+		if len(raw) > (C.sizeof_long) {
+			return 0, fmt.Errorf(
+				"length of bytes received from sysctl (%d) is greater than expected bytes (%d)",
+				len(raw),
+				C.sizeof_long,
+			)
+		}
+
+		if len(raw) < (C.sizeof_long) {
```

Comment:
I'd feel a little better if we checked explicitly to ensure it matches the size of an integer, and then threw an error otherwise.

Reply:
Thanks, I feel better about this too.